resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -216,8 +216,70 @@ private[spark] class Client(
                 .endContainer()
               .endSpec()
             .done()
-          submitCompletedFuture.get(30, TimeUnit.SECONDS)
-        }
+          try {
+            submitCompletedFuture.get(30, TimeUnit.SECONDS)
```

Comment:
IIRC the semantics here should be the same as that for YARN cluster-mode, which runs into a similar issue if resources aren't available for the application master. Otherwise we could support a fire and forget "wait forever" mode of operation, but that can come in a later iteration once we expand the spark-submit things (if/when we include killing jobs etc. possibly also)